semgrep/semgrep/metric_manager.py

```diff
@@ -151,6 +152,12 @@ def set_rules_with_findings(
     ) -> None:
         self._rules_with_findings = {r.full_hash: len(f) for r, f in findings.items()}
 
+    def set_user_flags(self, user_flags: Dict[str, Any]) -> None:
```

Comment:
If possible can we not have metric_manager store and then send arbitrary dict. It's probably less error prone to be more explicit here.

Reply:
You mean like having a metric_manager field for each flag?

Reply:
That might be a lot. Maybe what nathan suggested above and an allow-list of options and only copying those over? (Maybe even having a typed object be the arg here with allowed fields?)

Reply:
+1 to having a typed arg

Reply:
For which metrics: I think you can just copy all the ones we currently have. I don't think any of them are sensitive, as long as we only store whether they are used and not what argument they have